sequencer/src/lib.rs

```diff
@@ -286,53 +293,64 @@ pub async fn init_node(
     let public_ip = IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1));
     let orchestrator_client = OrchestratorClient::new(validator_args, public_ip.to_string());
 
-    let (mut config, wait_for_orchestrator) = match persistence.load_config().await? {
+    let private_staking_key = network_params.private_staking_key;
+    let public_staking_key = BLSPubKey::from_private(&private_staking_key);
+
+    let (config, wait_for_orchestrator) = match persistence.load_config().await? {
         Some(config) => {
             tracing::info!("loaded network config from storage, rejoining existing network");
             (config, false)
         }
         None => {
             tracing::info!("loading network config from orchestrator");
-            let config = orchestrator_client.get_config(public_ip.to_string()).await;
+            let mut config: NetworkConfig<VerKey, StaticElectionConfig> =
+                orchestrator_client.get_config(public_ip.to_string()).await;
+
+            // Get updated config from orchestrator containing all peer's public keys
+            config = orchestrator_client
+                .post_and_wait_all_public_keys(config.node_index, public_staking_key)
+                .await;
```

Comment:
Where is this `public_staking_key` assigned?  I think here it should be `config = orchestrator_client
                .post_and_wait_all_public_keys(config.node_index, config.config.my_own_validator_config.public_key)`.

Reply:
The two are the same.

https://github.com/EspressoSystems/espresso-sequencer/blob/78dfc4cd5f170626f51e9ce18e63e257eb86492f/sequencer/src/lib.rs#L242-L261